### PR TITLE
Develop-DG

### PR DESCRIPTION
**Pull request by MagnificentCreature**
This is a draft PR so you guys can see what Im changing, I can pull to @lilozz2 branch first, or merge onto another branch first before going to main

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -4,24 +4,40 @@ skinparam arrowThickness 1.1
 skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
-AddressBook *-right-> "1" UniquePersonList
```
Comment (lilozz2) [inline docs/diagrams/BetterModelClassDiagram.puml]: I think would be good to leave the UniquePersonList in

File: docs/images/BetterModelClassDiagram.png
Comment (KamiliArsyad) [inline docs/images/BetterModelClassDiagram.png:1]: Can we make this bigger

**Review by lilozz2 — COMMENTED** [review]
Good job, just a small thing on the BetterModelClassDiagram

File: docs/images/ModelClassDiagram.png
Comment (KamiliArsyad) [inline docs/images/ModelClassDiagram.png:1]: Make dis bigger

File: docs/diagrams/InteractionSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+' CommandResult --> InteractionCommand
+' deactivate CommandResult
+
+' InteractionCommand --> LogicManager : result
+' deactivate InteractionCommand
+
+' [<--LogicManager
+' deactivate LogicManager
+' @enduml
```
Comment (KamiliArsyad) [inline docs/diagrams/InteractionSequenceDiagram.puml]: Can we remove this

**Review by KamiliArsyad — APPROVED** [review]
need some slight changes on the diagram size

File: docs/diagrams/InteractionSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+deactivate Person
+Command --> Parser : return success
+deactivate Command
+Parser --> User : return success
+deactivate Parser
+User -> UI : update()
+activate UI
+UI --> User : display interaction
+deactivate UI
+@enduml
```
Comment (lilozz2) [inline docs/diagrams/InteractionSequenceDiagram.puml]: Would be good to follow the Class colours and not sure why the Classes are at the top and the bottom at the same time
image

File: docs/images/InteractionClassDiagram.png
Comment (KamiliArsyad) [inline docs/images/InteractionClassDiagram.png:1]: Should `date` and `interactionNote` be separated to compose interaction?

**Comment by KamiliArsyad** [comment]
Also uh, can you also put the `png` files along with the new PUML files

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 1 above)
Comment (MagnificentCreature) [reply docs/diagrams/BetterModelClassDiagram.puml]: Ok

**Review by KamiliArsyad — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -155,25 +156,32 @@ This section describes some noteworthy details on how certain features are imple
...
-The Interaction feature is implemented using the `Interaction` class. It is a simple class that contains the following fields:
+The Interaction feature is implemented using the `Interaction` class. A `Person` has a list of interactions and interactions have the following fields:
 
 ![InteractionClassDiagram](images/InteractionClassDiagram.png)
 
-InteractionNote will store the details of the interaction.
-InteractionOutcome is an Enumeration with the following values: Interested, NotInterested, FollowUpRequired, Unknown and Closed. They are used to store the outcome of the interaction.
-There is also a date field to store the date of the interaction. This is set to the current date when the interaction is created.
+`InteractionNote` will store the details of the interaction.
+`InteractionOutcome` is an Enumeration used to store the outcome of the interaction.
```
Comment (zhyuhan) [inline docs/DeveloperGuide.md]: Shall we match this to the name of the actual enum, `Outcome`?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -155,25 +156,32 @@ This section describes some noteworthy details on how certain features are imple
...
+There is also a `date` field to store the date of the interaction. This is set to the current date when the interaction is created.
 
-The `Person` class will contain a list of `Interaction` objects, which will be used to store the interactions with the client.
+When the user executes the `interaction` command, an `Interaction` is created and added to the `Person`'s list of interactions. The new `Interaction` is then displayed in the `PersonCard` automatically.
 
-The class also has getters and setters for the fields, as well as a `toString()` method to display the interaction in a readable format.
+Interactions play a key role in the reminder feature, recall that a follow-up is made X weeks after the last interaction. The `Interaction` class has a `getFollowUpDate()` method that is used to calculate the date of the next follow up for the `Person`. It does this by getting the latest interaction date and adding the follow up period of the `Person` to it.
 
-#### Design Considerations
+```
```
Comment (zhyuhan) [inline docs/DeveloperGuide.md]: Do we need to include this?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -320,6 +245,9 @@ Step 4. `UniqueReminderList` should also be updated daily with `ReminderSchedule
     * Pros: Reminders will always be up to date
     * Cons: Will require more processing power and might slow down the application
 
+### \[Proposed\] Data archiving
```
Comment (zhyuhan) [inline docs/DeveloperGuide.md]: Shall we remove this, since we are not going to write anything for this?

**Review by zhyuhan — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (MagnificentCreature) [reply docs/DeveloperGuide.md]: why not

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (MagnificentCreature) [reply docs/DeveloperGuide.md]: Okay!

**Pull request merged**